pkg/storage/chunk/client/azure/blob_storage_client.go

```diff
@@ -220,13 +220,17 @@ func NewBlobStorage(cfg *BlobStorageConfig, metrics BlobStorageMetrics, hedgingC
 func (b *BlobStorage) Stop() {}
 
 func (b *BlobStorage) ObjectExists(ctx context.Context, objectKey string) (bool, error) {
-	exists, _, err := b.ObjectExistsWithSize(ctx, objectKey)
-	return exists, err
+	_, err := b.objectAttributes(ctx, objectKey, "azure.ObjectExists")
+	return err == nil, err
 }
 
-func (b *BlobStorage) ObjectExistsWithSize(ctx context.Context, objectKey string) (bool, int64, error) {
+func (b *BlobStorage) GetAttributes(ctx context.Context, objectKey string) (client.ObjectAttributes, error) {
+	return b.objectAttributes(ctx, objectKey, "azure.GetAttributes")
+}
+
+func (b *BlobStorage) objectAttributes(ctx context.Context, objectKey, source string) (client.ObjectAttributes, error) {
```

Comment:
nit: IIUC using a different `source` is just going to create a different metric? Does it make sense to separate calls to this function into separate metrics? 
My suggestion would be to just drop the `source` parameter, if I'm understanding the code correctly

Reply:
Yes, the idea is to allow disambiguation between `GetAttributes` and `ObjectExists` where it is possible to do it. Loki isn't using `GetAttributes` so this shouldn't cause any problems but if you're using Loki's API for doing something, having this separation will help your application separating what is leading the object storage requests.